java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -792,7 +792,14 @@ static ChunkReader create(BufferChunk from, BufferChunk to) {
         current = (BufferChunk) current.next;
       }
       reqBytes += ef - cf;
-      return new ChunkReader(from, to, (int) (e - f), reqBytes);
+      if (reqBytes >= IOUtils.MAX_ARRAY_SIZE) {
+        throw new IllegalArgumentException("invalid reqBytes value " + reqBytes + ",out of bounds " + IOUtils.MAX_ARRAY_SIZE);
```

Comment:
UT don't have a use case for throwing an IllegalArgumentException, can you add it?

Reply:
Thank you for the reminder. I will refine it later

Reply:
I have added new UT related to readBytes and reqBytes validation.